src/ln/channelmonitor.rs

```diff
@@ -1123,36 +1185,41 @@ impl ChannelMonitor {
 					htlc_success_tx.input[0].witness[2].push(SigHashType::All as u8);
 
 					htlc_success_tx.input[0].witness.push(payment_preimage.to_vec());
-					htlc_success_tx.input[0].witness.push(chan_utils::get_htlc_redeemscript_with_explicit_keys(htlc, &local_tx.a_htlc_key, &local_tx.b_htlc_key, &local_tx.revocation_key).into_bytes());
+					let redeemscript = chan_utils::get_htlc_redeemscript_with_explicit_keys(htlc, &local_tx.a_htlc_key, &local_tx.b_htlc_key, &local_tx.revocation_key);
```

Comment:
This isnt the right redeem script - this is the one that corresponds with the input, but the spendable output is the one that corresponds with the output ie get_revokeable_redeemscript.

Reply:
Ah yes thanks! corrected